src/jsutils/instanceOf.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+/**
+ * A replacement for instanceof which includes an error warning when multi-realm
+ * constructors are detected.
+ */
+declare function instanceOf(
+  value: mixed,
+  constructor: mixed,
+): boolean %checks(value instanceof constructor);
+
+// eslint-disable-next-line no-redeclare
+export default function instanceOf(value, constructor) {
+  if (value instanceof constructor) {
+    return true;
+  }
+  if (value) {
+    const valueConstructor = value.constructor;
+    if (valueConstructor && valueConstructor.name === constructor.name) {
+      throw new Error(
+        `Cannot use ${constructor.name} "${value}" from another module or realm.
+
+Ensure that there is only one instance of "graphql" in the node_modules
```

Comment:
@leebyron Not sure if anybody still using alternative package managers(e.g. bower) but maybe it worth to prefix this string with "If you are using NPM package" to reduce possible confusion.